### PR TITLE
add grammar support for @modifier.

### PR DESCRIPTION
**Pull request by yu-shipit**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :**
Filodb does not handle [@modifier](https://prometheus.io/docs/prometheus/latest/querying/basics/#modifier). 

**New behavior :**
filodb will be able to parse  [@modifier](https://prometheus.io/docs/prometheus/latest/querying/basics/#modifier).  
will support query parsing like
```
foo @timestamp,  foo @start(),  foo @end()
foo[1m] @timestamp,  foo[1m] @start(),  foo[1m] @end() 
rate(foo[1m] @timestamp),  rate(foo[1m] @start()),  rate((foo[1m] @end())
```

```
foo @timestamp offset 1d, foo offset 1d @timestamp
foo[1m] @timestamp offset 1d, foo[1m] offset 1d @timestamp
rate(foo[1m] @timestamp offset 1d), rate(foo[1m] offset 1d @timestamp)
```

Note that the an at modifier can coexist with an offset modifier irrespective of order.

Breaking changes may include:
- Changes to query parsing / PromQL parsing

### CODE REVIEW

File: prometheus/src/main/java/filodb/prometheus/antlr/PromQL.g4 [hunk 1]
```diff
@@ -55,6 +55,10 @@ window: '[' DURATION ']';
 
 offset: OFFSET DURATION;
 
+atModifier: AT(NUMBER | START | END);
+
+modifier: offset | atModifier | offset atModifier | atModifier offset;
```
[the remark this answers is not in the document]
Comment (alextheimer) [reply prometheus/src/main/java/filodb/prometheus/antlr/PromQL.g4:60]: https://prometheus.io/docs/prometheus/latest/querying/basics/#modifier

**Review by alextheimer — CHANGES_REQUESTED** [review]
LGTM, just one small update requested in the comments-- please feel free to dismiss this review after that's updated.

**Review by alextheimer — DISMISSED** [review]
Approved for the sake of forward progress -- please feel free to merge after the other comments are addressed.

File: prometheus/src/main/java/filodb/prometheus/antlr/PromQL.g4 (hunk 1 above)
[the remark this answers is not in the document]
Comment (yu-shipit) [reply prometheus/src/main/java/filodb/prometheus/antlr/PromQL.g4:60]: good catch. It seems the white space is optional.
http://demo.robustperception.io:9090/graph?g0.expr=topk(5%2C%20rpc_durations_histogram_seconds_bucket%7Binstance%3D%22demo.do.prometheus.io%3A8996%22%7D%20%40start())&g0.tab=0&g0.stacked=0&g0.range_input=1d&g1.expr=%20max(rpc_durations_histogram_seconds_bucket%7Binstance%3D%22demo.do.prometheus.io%3A8996%22%7D)%20&g1.tab=0&g1.stacked=0&g1.range_input=1h&g2.expr=rpc_durations_histogram_seconds_bucket%7Binstance%3D%22demo.do.prometheus.io%3A8996%22%2C%20job%3D%22random%22%2C%20le%3D%22%2BInf%22%7D&g2.tab=0&g2.stacked=0&g2.range_input=1h

[the remark this answers is not in the document]
Comment (yu-shipit) [reply prometheus/src/main/java/filodb/prometheus/antlr/PromQL.g4:60]: It seems the parser can naturally handle both cases. added @ start() and @start() to the tests.

**Review by alextheimer — APPROVED** [review]
Thanks for the additional tests @yu-shipit 🙏

**Pull request merged**